src/js/controllers/step1/LottoController.js

```diff
@@ -0,0 +1,219 @@
+    if (isNaN(number)) throw new Error("번호는 숫자 타입이어야 합니다.");
+
+    if (number < 1 || number > 45) throw new Error("번호는 1 이상 45 이하 숫자여야 합니다.");
+  }
+
+  validateWinningNumbers(winningNumbers) {
+    winningNumbers.forEach((number) => {
+      this.validateNumber(number);
+    });
+
+    if (winningNumbers.length !== 6) {
+      throw new Error("당첨 번호는 6개의 숫자여야 합니다.");
+    }
+
+    if (new Set(winningNumbers).size !== winningNumbers.length) {
+      throw new Error("당첨 번호는 중복되면 안됩니다.");
+    }
+  }
+
+  validateBonusNumber(winningNumbers, bonusNumber) {
+    this.validateNumber(bonusNumber);
+
+    if (winningNumbers.indexOf(bonusNumber) !== -1)
+      throw new Error("보너스 번호는 당첨 번호와 중복되면 안됩니다.");
+  }
+
+  async startLotto() {
+    const money = await this.readMoney();
+    this.#money = money;
+    const lottoTickets = this.issueLottoTickets(money);
+
+    this.printLottoTickets(lottoTickets);
+
+    const { winningNumbers, bonusNumber } = await this.readNumbers();
+    this.printWinningStatistics({ lottoTickets, winningNumbers, bonusNumber });
+
+    this.#readline.close();
+  }
+
+  async readMoney() {
+    const money = await this.#consoleInput.readline("구입금액을 입력해 주세요.");
+    this.validateMoney(money);
+
+    return Number(money);
+  }
+
+  async readNumbers() {
+    const winningNumbersStr = await this.#consoleInput.readline("당첨 번호를 입력해주세요.");
+
+    const winningNumbers = winningNumbersStr.split(",").map(Number);
+    this.validateWinningNumbers(winningNumbers);
+
+    const bonusNumber = Number(await this.#consoleInput.readline("보너스 번호를 입력해 주세요."));
+    this.validateBonusNumber(winningNumbers, bonusNumber);
+
+    return { winningNumbers, bonusNumber };
+  }
+
+  issueLottoTickets(money) {
+    const lottoTickets = [];
+    const NumberOfLotto = Math.floor(money / LottoTicket.price);
+
+    for (let i = 0; i < NumberOfLotto; i++) {
+      lottoTickets.push(new LottoTicket(this.drawLottoNumbers()));
+    }
+
+    return lottoTickets;
+  }
+
+  drawLottoNumbers() {
+    const numbers = [];
+
+    while (numbers.length < 6) {
+      const randomNumber = Math.floor(Math.random() * 45) + 1;
+      if (numbers.indexOf(randomNumber) === -1) numbers.push(randomNumber);
+    }
+
+    return numbers.sort((a, b) => a - b);
+  }
+
+  exportStatistics({ lottoNumbers, winningNumbers, bonusNumber }) {
+    const count = this.countMatchNumbers({
+      lottoNumbers,
+      winningNumbers,
+      bonusNumber,
+    });
+
+    this.#winningCriteria.forEach((criteria) => {
+      if (
+        criteria.winningCount === count.winningCount &&
+        (count.winBonus || !criteria.hasToWinBonus)
+      )
+        criteria.numOfWinTicket++;
+    });
+  }
+
+  countMatchNumbers({ lottoNumbers, winningNumbers, bonusNumber }) {
+    const count = {
+      winningCount: 0,
+      winBonus: false,
+    };
+
+    count.winningCount = this.countWinningNumbers(lottoNumbers, winningNumbers);
+
+    if (lottoNumbers.indexOf(bonusNumber) !== -1) count.winBonus = true;
```

Comment:
index가 사용되지 않고 값이 있는지 여부를 판단하는거라면 `includes()` 메서드를 활용해보셔도 좋을거 같네요!

```suggestion
    if (lottoNumbers.includes(bonusNumber)) count.winBonus = true;
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes